### ISSUE TITLE
Registrations/Logins/Roles/Authorized Redirect URLs are "gone" after refresh

### ISSUE THREAD

**Issue by ceefour** [FusionAuth/fusionauth-issues#563]
### Description 

This is probably multiple bugs in one, but the symptoms are similar.
Let me know if you want to split this into multiple bug reports.

First my disclaimer is that I'm using MariaDB, specifically AWS RDS MariaDB.
I'm aware that this is not 100% supported yet, but I'm hoping that **if** the issues turn out to be  caused by MariaDB, then I hope this bug report will help work on #327 and #367.

### Steps to reproduce

* In Users > (choose user) > Manage : Last login is "-", even if Session tab shows that user has logged in using OAuth 
OpenID Connect with that specific application.

* In Users > (choose user) > Manage > Registrations. After adding a registration, then wait about 1-3 minutes, refresh, the registration is gone.

* In Users > (choose user) > Manage > Registrations: Last login is "-", even if Session show that user has logged in.

* In Users > Applications > (choose application) > Manage Roles: After adding a role, then wait about 1-3 minutes, refresh, the role is "gone". However, the role is actually "still there", just not displayed. We can check this by adding a new role with the same name, which will error with message "Exists".

* In Applications > (choose application) > Edit > Set "Authorized Redirect URLs": After adding an URL, then Save. Then edit again, the URL is "gone".

Note: `kubectl logs -f fusionauth-*` show no logging at all during these operations.

### Expected behavior

The data is still there and displayed.

### Workaround

Restarting the FusionAuth server (or deleting the pod and let deployment bring the replica back) does make the "gone" data "come back".

### Screenshots

Example of "disappearing" Role:

![image](https://user-images.githubusercontent.com/24123/79653197-1a2a7600-81dc-11ea-8216-aa4e92e6d5e8.png)

### Platform

Server:

- DigitalOcean
- Kubernetes 1.16
- Elasticsearch 6.8.8
- AWS RDS MySQL 8.0.17 (previously on AWS RDS MariaDB 10.3 and 10.4)
- FusionAuth 1.16.0-RC.1 installed using helm chart (previously 1.15.8)

Client:

 - Device: Desktop
 - OS: Ubuntu 18.04
 - Browser + version: Chrome 81

### Additional context

I'm willing to help diagnose by sending logs, turning on debugging, etc. but I'm very new so any instructions will be very helpful.

Journal: https://about.lovia.life/handbook/technology/fusionauth/

**Comment by ceefour** [comment]
Was suspecting this was due MariaDB `max_connections` setting, however after increasing the setting from 40 to 150 (which should be more than plenty), problem still persists, this was not the cause.

(also: I would expect connection errors to write error logs)

**Comment by robotdan** [comment]
MariaDB is not supported, so your mileage may vary.

**Comment by ceefour** [comment]
@robotdan I'm aware of it. Can you share pointers/hints to enable better debugging/diagnostics?

What I'm looking for is currently not a solution, but 
with your help if I can pinpoint specific problems and causes, that would be good progress. After that the next step to resolve it or just document it so that at least users are aware of specific issues.

**Comment by ceefour** [comment]
I tried to check character set and collation and it turned out I used character set `utf8mb4` and collation `utf8mb4_bin`. So I changed the database and all tables to use `utf8mb4_unicode_ci` collation. There's no change in behavior, so this was not the issue.

(I'm also not sure yet whether MariaDB is the problem or not)

**Comment by voidmain** [comment]
@ceefour - I'm guessing this is a MariaDB issue and you can prove that pretty easily if you want to give it a try. Here's what you need to do:

1. Install PostgreSQL 11.4
2. Create a super user in your PostgreSQL database named `postgres` with a password of `password`
3. Create an ordinary user in your PostgreSQL database named `fusionauth` with a password of `fusionauth`
4. Pre-load the database schema manually by following the instructions here: https://fusionauth.io/docs/v1/tech/installation-guide/fusionauth-app#advanced-installation
5. Keep your installation of FusionAuth exactly the same and don't change anything EXCEPT for the database configuration in `fusionauth.properties`. Point FusionAuth at your new PostgreSQL database by changing the `database.url`, `database.username` and `database.password`. 

After you have completed those steps, try to reproduce your issues and if they don't happen any longer, then you have confirmed it is MariaDB. If they occur again, then we can start looking into them.

**Comment by ceefour** [comment]
Thanks very much for your care @voidmain :+1: 

Since the intention is to keep it "exactly the same":

* After step 4, there would be another step to load the previous data into (Postgre?)SQL.. otherwise all previous data would already be gone during the switch?
* I assume I can also try it with MySQL instead of PostgreSQL, since MySQL Is a supported configuration... and I can dump/restore my data from MariaDB into MySQL. Is this correct?

**Comment by ceefour** [comment]
Since the helm release upgrade restarted the pod, it seems that I'm now already on `1.16.0-RC.1` not `1.15.8` anymore.

I'd like to confirm, is it intentional that the Docker tag `latest` refers to latest pre-release versions as well? I'd expect the convention to be:

* `latest`: latest stable
* `develop`: pre-release versions (if available) or latest stable (if no pre-release newer than stable)

If not, I can file another issue discussing the docker tags mechanism. For the least surprise principle, I'd recommend `latest` to only refer to stable versions. However, if pre-release versions can still be referred by `latest`, the alternative is to have another tag e.g. `stable` that refers only latest stable versions.

**Comment by ceefour** [comment]
![image](https://user-images.githubusercontent.com/24123/80629836-9430e980-8a7d-11ea-882a-9a8887428190.png)

WOOOHOOOOOOOOOOOOOOOO 🕺🕺🕺🕺🕺🕺

**Comment by ceefour** [comment]
I'm now on MySQL 8.0.17 (which is a supported configuration), and the problem still persists.

I suspected it was due to me using Elasticsearch 6.8.8 instead of 6.3.1, however I unintentionally upgraded to FusionAuth 1.16.0-RC.1, which has Elasticsearch 7 support.

As currently this happening on FusionAuth remote instance which is being used daily, I will try to reproduce this issue by using local docker entirely (with fusionauth+mysql+elasticsearch docker). If I manage to reproduce that consistently, I plan to share the exact docker containers snapshot to make it easy for you to quickly run it there.

**Comment by robotdan** [comment]
@ceefour this is a good point, we will update our tagging strategy to ensure we only publish a latest tag for a stable release.

**Comment by ceefour** [comment]
@robotdan Ok I've filed the docker image tag request at #596

**Comment by robotdan** [comment]
Are you able to recreate on the latest stable release 1.15.8?

**Comment by ceefour** [comment]
@robotdan if you mean whether it happens on 1.15.8, yes the bug originally using 1.15.8+MariaDB, then still persists in 1.15.8+MySQL.

after 2 days in 1.16.0.RC1 I am now back to 1.15.8+MySQL.

I haven't yet isolated the docker containers though.
